src/store/actions/filesActions.js

```diff
@@ -4,26 +4,31 @@ import {
   FILES_SET_FILES_LIST,
   FILES_SET_FOLDER_LIST,
   FILES_ADD_CACHED_FILES,
+  FILES_SET_IS_LOADING,
 } from '../constants';
 
 /* SYNC OPERATIONS */
 
 export const setFilesList = (payload) => ({ type: FILES_SET_FILES_LIST, payload });
 export const setFoldersList = (payload) => ({ type: FILES_SET_FOLDER_LIST, payload });
 export const setCachedFiles = (payload) => ({ type: FILES_ADD_CACHED_FILES, payload });
+export const setFilesLoading = (payload) => ({ type: FILES_SET_IS_LOADING, payload });
 
 /* ASYNC OPERATIONS */
 
 export const fetchFileListFromPath = (path) => (dispatch, getState) => {
   const token = getState().app.get('token');
+  dispatch(setFilesLoading(true));
 
   apiFetchFiles(token, path)
```

Comment:
```jsx
  apiFetchFiles(token, path)
     .then(({ data }) => {
       dispatch(setFoldersList(filterAndSortFolders(data)));
       dispatch(setFilesList(filterAndSortSongs(data)));
       dispatch(setCachedFiles({ path, files: data }));
     })
     .catch((err) => {
       throw new Error(err);
     })
    .finally(() => {
       dispatch(setFilesLoading(false));
    });
```

Reply:
This seems weird, can't we use a single transform for the data and handle the filter/sort of files, song and caching there? Like a service

Reply:
That would be the ideal. Sadly, the API is not well designed so files & songs don't share the same object's shape.